game/world/triggers/movetrigger.h

```diff
@@ -39,23 +30,42 @@ class MoveTrigger : public AbstractTrigger {
       };
 
     struct KeyLen {
-      uint64_t ticks    = 0;
+      uint64_t ticks = 0;
       };
 
+    void   moveEvent() override;
+
+    void   preProcessTrigger(State prev = Idle);
+    void   postProcessTrigger();
+    size_t nextFrame(size_t i) const;
+    size_t prevFrame(size_t i) const;
+
+    void   setView     (MeshObjects::Mesh&& m);
+    void   emitSound   (std::string_view snd, bool freeSlot=true) const;
+    void   advanceAnim ();
+    void   advanceKey  (uint64_t dt);
+
+    void   invalidateView();
+
     Tempest::Matrix4x4                   pos0;
     MeshObjects::Mesh                    view;
     PhysicMesh                           physic;
     std::vector<KeyLen>                  keyframes;
     std::vector<zenkit::AnimationSample> moverKeyFrames;
     zenkit::MoverBehavior                behavior;
+    zenkit::MoverSpeedType               speedType;
+    zenkit::MoverLerpType                lerpType;
+    bool                                 autoRotate;
     std::string                          sfxOpenStart;
     std::string                          sfxOpenEnd;
+    std::string                          sfxCloseStart;
     std::string                          sfxCloseEnd;
     std::string                          sfxMoving;
     std::string                          visualName;
     float                                stayOpenTimeSec;
 
     State                                state     = Idle;
     uint64_t                             sAnim     = 0;
-    uint32_t                             frame     = 0;
+    size_t                               frame     = 0;
+    float                                alpha     = 0;
```

Comment:
Is it required no to track alpha persistently (and serialize it)?

Reply:
`alpha` now tells us where the mover is located between two keyframes. `sAnim` solely counts time since a mover is open for `OpenTime `movers.
We could as well use `frame` instead on load since it's already serialized and now updated once a new keyframe is reached? Should be good enough and mostly relevant for looping movers anyway.

Reply:
Yeah, I think it's a good approach to use only `frame` and derive `alpha`(linear) and `a`(attenuated alpha) during the tick, as local variable.
This should make the code more readable.

Reply:
I changed `sAnim` to be time since last `frame` change so alpha is not needed anymore.